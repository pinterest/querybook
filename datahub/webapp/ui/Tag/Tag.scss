.Tag,
.TagGroup {
    display: inline-block;
    margin: 2px 4px;
    border-radius: var(--border-radius);
}

.Tag {
    background-color: var(--tag-bg-color);
    padding: 2px 8px;
    color: var(--tag-text-color);
    font-weight: var(--bold-font);
    font-size: var(--small-text-size);

    &.highlighted {
        background-color: var(--color-accent-bg);
    }
}

.TagGroup {
    overflow: hidden;

    .Tag {
        margin: 0;
        border-radius: 0;
    }
}
