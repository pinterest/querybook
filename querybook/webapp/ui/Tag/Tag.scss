@mixin small-tag() {
    font-size: var(--xxsmall-text-size);
    padding: 2px 4px;
}

.Tag,
.TagGroup {
    display: inline-flex;
    margin: 2px 4px;
    border-radius: var(--border-radius);
    cursor: default;
}

.Tag {
    background-color: var(--tag-bg-color);
    padding: 2px 8px;
    color: var(--tag-text-color);
    font-weight: var(--bold-font);
    font-size: var(--small-text-size);

    &.highlighted {
        background-color: var(--color-accent-bg);
    }
    .HoverIconTag-hover {
        display: none;
        .Icon {
            padding-left: 4px;
            svg {
                height: 16px;
                width: 16px;
            }
        }
    }
    &:hover {
        .HoverIconTag-hover {
            display: flex;
            cursor: pointer;
        }
    }

    &.small {
        @include small-tag();
    }
}

.TagGroup {
    overflow: hidden;

    .Tag {
        margin: 0;
        border-radius: 0;
    }

    &.small .Tag {
        @include small-tag();
    }
}
