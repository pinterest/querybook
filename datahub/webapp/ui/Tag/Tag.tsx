import React from 'react';

import './Tag.scss';

export interface ITagGroupProps {
    className?: string;
    children: React.ReactNode;
}

export interface ITagProps {
    children: React.ReactNode;
    highlighted?: boolean;
}

export const TagGroup: React.FunctionComponent<ITagGroupProps> = ({
    className,
    children,
}) => {
    return <div className={`${className} TagGroup`}>{children}</div>;
};

export const Tag: React.FunctionComponent<ITagProps> = ({
    children,
    highlighted = false,
}) => {
    return (
        <span className={highlighted ? 'Tag highlighted' : 'Tag'}>
            {children}
        </span>
    );
};
