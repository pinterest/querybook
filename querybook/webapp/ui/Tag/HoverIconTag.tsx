import clsx from 'clsx';
import React from 'react';

import { Icon } from 'ui/Icon/Icon';
import type { AllLucideIconNames } from 'ui/Icon/LucideIcons';

import { ITagProps, Tag } from './Tag';

export interface IHoverIconTagProps extends ITagProps {
    iconOnHover?: AllLucideIconNames;
    onIconHoverClick?: (e?: React.MouseEvent) => any;
}
export const HoverIconTag = React.forwardRef<
    HTMLSpanElement,
    IHoverIconTagProps
>(({ iconOnHover, onIconHoverClick, children, ...tagProps }, ref) => {
    const hoverDOM = iconOnHover ? (
        <div className="HoverIconTag-hover" onClick={onIconHoverClick}>
            <Icon name={iconOnHover} />
        </div>
    ) : null;

    const className = clsx(tagProps['className'], 'HoverIconTag');

    return (
        <Tag {...tagProps} ref={ref} className={className}>
            {children}
            {hoverDOM}
        </Tag>
    );
});
