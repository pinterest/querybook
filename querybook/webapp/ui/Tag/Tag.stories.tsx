import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import centered from '@storybook/addon-centered/react';

import { Tag, TagGroup } from './Tag';

export const _Tag = () => (
    <>
        <TagGroup>
            <Tag>First</Tag>
            <Tag highlighted>Second</Tag>
        </TagGroup>
        <Tag>Tag</Tag>
        <Tag highlighted>Highlighted Tag</Tag>
    </>
);

export default {
    title: 'Stateless/Tag',
    decorators: [centered],
};
