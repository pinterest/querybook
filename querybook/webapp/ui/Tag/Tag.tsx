import React from 'react';
import clsx from 'clsx';

import { TooltipDirection } from 'const/tooltip';

import { Icon } from 'ui/Icon/Icon';

import './Tag.scss';

export interface ITagGroupProps {
    className?: string;
    children: React.ReactNode;
}

export interface ITagProps {
    children: React.ReactNode;

    highlighted?: boolean;

    tooltip?: React.ReactNode;
    tooltipPos?: TooltipDirection;

    onClick?: () => any;

    className?: string;
}
export interface IHoverIconTagProps extends ITagProps {
    iconOnHover?: string;
    onIconHoverClick?: () => any;
}

export const TagGroup: React.FunctionComponent<ITagGroupProps> = ({
    className,
    children,
}) => <div className={`${className} TagGroup`}>{children}</div>;

export const Tag: React.FunctionComponent<ITagProps> = ({
    children,
    highlighted = false,
    tooltip,
    tooltipPos = 'top',
    onClick,
    className,
}) => {
    const tooltipProps = {};
    if (tooltip) {
        tooltipProps['aria-label'] = tooltip;
        tooltipProps['data-balloon-pos'] = tooltipPos;
    }

    const tagClassname = clsx({
        Tag: true,
        'flex-row': true,
        highlighted,
        [className]: Boolean(className),
    });

    return (
        <span {...tooltipProps} onClick={onClick} className={tagClassname}>
            {children}
        </span>
    );
};

export const HoverIconTag: React.FunctionComponent<IHoverIconTagProps> = ({
    iconOnHover,
    onIconHoverClick,
    children,
    ...tagProps
}) => {
    const hoverDOM = iconOnHover ? (
        <div className="HoverIconTag-hover" onClick={onIconHoverClick}>
            <Icon name={iconOnHover} />
        </div>
    ) : null;

    tagProps['className'] = (tagProps['className'] ?? '') + ' HoverIconTag';

    return (
        <Tag {...tagProps}>
            {children}
            {hoverDOM}
        </Tag>
    );
};
