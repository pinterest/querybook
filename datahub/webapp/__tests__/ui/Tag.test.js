import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Tag } from '../../ui/Tag/Tag';

it('renders without crashing', () => {
    shallow(<Tag>test</Tag>);
});

describe('matches enzyme snapshots', () => {
    it('matches snapshot', () => {
        const wrapper = shallow(<Tag>test</Tag>);
        const serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
    it('matches snapshot', () => {
        const wrapper = shallow(<Tag highlighted>test</Tag>);
        const serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
});
