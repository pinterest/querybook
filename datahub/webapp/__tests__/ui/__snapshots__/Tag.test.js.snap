// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches enzyme snapshots matches snapshot 1`] = `
<span
  className="Tag flex-row undefined"
>
  test
</span>
`;

exports[`matches enzyme snapshots matches snapshot 2`] = `
<span
  className="Tag flex-row highlighted undefined"
>
  test
</span>
`;
